src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/ISimdVector_2.cs

```diff
@@ -0,0 +1,450 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Numerics;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+namespace System.Runtime.Intrinsics
+{
+    /// <summary>Defines a single instruction, multiple data (SIMD) vector type.</summary>
+    /// <typeparam name="TSelf">The type that implements the interface.</typeparam>
+    /// <typeparam name="T">The type of the elements in the vector.</typeparam>
+    internal interface ISimdVector<TSelf, T>
+        : IAdditionOperators<TSelf, TSelf, TSelf>,
+          // IAdditiveIdentity<TSelf, TSelf>,
+          IBitwiseOperators<TSelf, TSelf, TSelf>,
+          // IComparisonOperators<TSelf, TSelf, bool>,
+          // IDecrementOperators<TSelf>,
+          IDivisionOperators<TSelf, TSelf, TSelf>,
+          IEqualityOperators<TSelf, TSelf, bool>,
+          IEquatable<TSelf>,
+          // IIncrementOperators<TSelf>,
+          // IMinMaxValue<TSelf>,
+          // IModulusOperators<TSelf, TSelf, TSelf>,
+          // IMultiplicativeIdentity<TSelf, TSelf>,
+          IMultiplyOperators<TSelf, TSelf, TSelf>,
+          // IShiftOperators<TSelf, int, TSelf>
+          // ISpanFormattable,
+          ISubtractionOperators<TSelf, TSelf, TSelf>,
+          IUnaryNegationOperators<TSelf, TSelf>,
+          IUnaryPlusOperators<TSelf, TSelf>
+        where TSelf : unmanaged, ISimdVector<TSelf, T>
+        where T : struct
```

Comment:
These may be "more restrictive" than necessary.

In the case of `where TSelf : unmanaged` it gives as rationale behavior for reinterpret casts and other behaviors so we can provide default implementations for several methods. Once it could contain a reference type (such as from a 3rd party type implementing these) many assumptions go out the window.

In the case of `where T : struct` it just matches the current constraint on the `Vector64/128/256<T>` types.